src/ssl.c

```diff
@@ -260,6 +262,541 @@ const WOLF_EC_NIST_NAME kNistCurves[] = {
 };
 #endif
 
+#if defined(HAVE_ECH)
+int wolfSSL_CTX_GenerateEchConfig(WOLFSSL_CTX* ctx, const char* publicName)
+{
+    int ret = 0;
+    int rngRet = 0;
+    word16 encLen = DHKEM_X25519_ENC_LEN;
+#ifdef WOLFSSL_SMALL_STACK
+    Hpke* hpke;
+    WC_RNG* rng;
+#else
+    Hpke hpke[1];
+    WC_RNG rng[1];
+#endif
+
+    if (ctx == NULL || publicName == NULL)
+      return BAD_FUNC_ARG;
+
+    ctx->echConfigs = (WOLFSSL_EchConfig*)XMALLOC(sizeof(WOLFSSL_EchConfig),
+        ctx->heap, DYNAMIC_TYPE_TMP_BUFFER);
+    ForceZero(ctx->echConfigs, sizeof(WOLFSSL_EchConfig));
+
+    if (ctx->echConfigs == NULL)
+        return MEMORY_E;
+
+#ifdef WOLFSSL_SMALL_STACK
+    hpke = (Hpke*)XMALLOC(sizeof(Hpke), ctx->heap, DYNAMIC_TYPE_TMP_BUFFER);
+
+    if (hpke == NULL)
```

Comment:
please match having the bracket on the same line as the if condition,

`if (hpke == NULL) {`